src/zenml/integrations/kubernetes/metadata_stores/kubernetes_metadata_store.py

```diff
@@ -0,0 +1,102 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of Kubernetes metadata store."""
+
+from typing import ClassVar
+
+from kubernetes import client, config
+
+from zenml.integrations.kubernetes import KUBERNETES_METADATA_STORE_FLAVOR
+from zenml.integrations.kubernetes.orchestrators.kube_utils import (
+    create_mysql_deployment,
+    create_namespace,
+    delete_deployment,
+    make_core_v1_api,
+)
+from zenml.logger import get_logger
+from zenml.metadata_stores.mysql_metadata_store import MySQLMetadataStore
+
+logger = get_logger(__name__)
+
+
+class KubernetesMetadataStore(MySQLMetadataStore):
+    """Kubernetes metadata store (MySQL database deployed in the cluster).
+
+    Attributes:
+        kubernetes_namespace: Name of the kubernetes namespace.
+            Defaults to "default".
+        deployment_name: Name of the deployment (and corresponding service/pod)
+            that will be created when calling `provision()`.
+            Defaults to "mysql".
+        storage_capacity: Storage capacity of the metadata store.
+            Defaults to `"10Gi"` (=10GB).
+    """
+
+    kubernetes_namespace: str = "default"
+    deployment_name: str = "mysql"
+    storage_capacity: str = "10Gi"
+
+    FLAVOR: ClassVar[str] = KUBERNETES_METADATA_STORE_FLAVOR
+
+    @property
+    def deployment_exists(self) -> bool:
+        """Check whether a MySQL deployment exists in the cluster.
+
+        Returns:
+            Whether a MySQL deployment exists in the cluster.
+        """
+        config.load_kube_config()
+        v1 = client.AppsV1Api()
+        resp = v1.list_namespaced_deployment(
+            namespace=self.kubernetes_namespace
+        )
+        for i in resp.items:
+            if i.metadata.name == self.deployment_name:
+                return True
+        return False
+
+    @property
+    def is_provisioned(self) -> bool:
```

Comment:
There is also the property `is_running` that you probably want to overwrite. It indicated whether the provisioned resources are actively running/reachable or if they're suspended, which in the current implementation isn't possible so it should probably return the same value as `is_provisioned`:
```python
@property
def is_running(self) -> bool:
    return self.is_provisioned
```

If it's not overwritten however it always returns `True`, which might or might not get us into trouble.